trie/validation.py

```diff
@@ -38,3 +42,10 @@ def validate_is_node(node):
                 validate_length(sub_node, 32)
     else:
         raise ValidationError("Invalid Node: {0}".format(node))
+
+
+def validate_is_bin_node(node):
+    if node == BLANK_HASH or node[0] in (BRANCH_TYPE, KV_TYPE, LEAF_TYPE):
```

Comment:
*small nitpick*

Lets move `(BRANCH_TYPE, KV_TYPE, LEAF_TYPE)` to a constant, something like `BINARY_TREE_NODE_TYPES` and change it to be a set.

Then lets expand this function for slightly easier readability.

```python
if node = BLANK_HASH:
    ....
elif node in BINARY_TREE_NODE_TYPES:
    ...
else:
    ...
```